builder/api/README.md

```diff
@@ -0,0 +1,173 @@
+| `Prototype.@Annotated`     | `false`  | Allows adding an annotation (or annotations) to the generated class or methods                                                                                        |
+| `Prototype.@FactoryMethod` | `false`  | Use in generated code to mark static factory methods, also can be used on blueprint factory methods to be used during code generation                                 |
+| `Prototype.@Singular`      | `false`  | Used for lists, sets, and maps to add methods `add*` in addition to the full collection setters                                                                       |     
+| `RuntimeType.@Prototype`   | `true`   | Annotation on runtime type that is created from a `Prototype`, to map it to the prototype it can be created from, used to trigger annotation processor for validation |
+
+Interfaces:
+
+| Interface                      | Generated | Description                                     |
+|--------------------------------|-----------|-------------------------------------------------|
+| `RuntimeType`                  | `false` | runtime type must implement this interface to mark which prototype is used to create it |
+| `Prototype.Factory`            | `false` | if blueprint implements factory, it means the prototype is used to create a single runtime type and will have methods `build` and `get` both on builder an on prototype interface that create a new instance of the runtime object |
+| `Prototype.BuilderInterceptor` | `false` | custom interceptor to modidfy buider before validation is done in method `build` |
+| `Prototype`                    | `true` | all prototypes implement this interface |
+| `Prototype.Builder`            | `true`    | all prototype builders implement this interface |
+
+
+## Configured providers
+We can define a configured option as follows:
+`@ConfiguredOption(key = "security-providers", provider = true, providerType = SecurityProviderProvider.class, providerDiscoverServices = false)`
+
+Rules:
+1. `providerType` MUST extend `io.helidon.common.config.ConfiguredProvider`
+2. The method MUST return a `List` of the type the provider creates, so in this case we consider the `SecurityProviderProvider`
+     to be capable of creating a `SecurityProvider` instance from configuration, so the return type would
+     be `List<SecurityProvider>`, where `SecurityProvider extends NamedService` and  
+     `SecurityProviderProvider extends ConfiguredProvider<SecurityProvider>`
+
+This will expect the following configuration:
+```yaml
+security-providers:
+  discover-services: true # optional, to override "providerDiscoverServices" option
+  providers:
+    - name: "my-provider"
+      type: "http-basic"
+      enabled: true
+```
+
+The generated code will read all nodes under `providers` and map them to an instance.  
+
+## Config driven
```

Comment:
nit. Let's settle on Config-driven as a convention, with a dash.

Reply:
Config driven removed from buidler API docs, added to config driven module

Reply:
pushed